src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -258,9 +258,27 @@ ss::future<try_abort_reply> tx_gateway_frontend::do_try_abort(
                     try_abort_reply{.aborted = true, .ec = tx_errc::none});
               }
               auto tx_id = tx_id_opt.value();
-              return stm->get_tx_lock(tx_id)->with(
-                [&self, stm, tx_id, pid, tx_seq, timeout]() {
-                    return self.do_try_abort(stm, tx_id, pid, tx_seq, timeout);
+
+              auto tx_opt = stm->get_tx_by_id(tx_id);
+              if (!tx_opt) {
+                  return ss::make_ready_future<try_abort_reply>(
+                    try_abort_reply{.aborted = true, .ec = tx_errc::none});
+              }
+              auto tx = tx_opt.value();
+              if (tx.pid != pid || tx.tx_seq != tx_seq) {
+                  return ss::make_ready_future<try_abort_reply>(
+                    try_abort_reply{.aborted = true, .ec = tx_errc::none});
+              }
+
+              return stm->get_tx_lock(tx_id)
+                ->with(
+                  timeout,
+                  [&self, stm, tx_id, pid, tx_seq, timeout]() {
+                      return self.do_try_abort(
+                        stm, tx_id, pid, tx_seq, timeout);
+                  })
+                .handle_exception_type([](const ss::semaphore_timed_out&) {
+                    return try_abort_reply{.ec = tx_errc::unknown_server_error};
```

Comment:
just to understand how this works: if .ec != none, then .ec takes precedence by the caller? I guess I would have expected there to be a .aborted = true here, too.

Reply:
aborted isn't a property of the request but a property of a transaction. when a coordinator answers `.aborted=true` it gives a signal that the transactions was aborted and it's save to move forward up. here a request times out on waiting for the lock, so we return an error (unknown_server_error)